### PR TITLE
API V3: clean version when deactivated and build version when activated

### PR DESCRIPTION
**Pull request by stsewd**
This matches what we do in our version form.

Closes https://github.com/readthedocs/readthedocs.org/issues/10221.

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--10308.org.readthedocs.build/en/10308/

- Developer's documentation (`dev`): https://dev--10308.org.readthedocs.build/en/10308/

### CODE REVIEW

File: readthedocs/api/v3/views.py [hunk 1]
```diff
@@ -297,6 +298,33 @@ def get_serializer_class(self):
...
+        # Get the updated version.
+        version = self.get_object()
+        # If the version is deactivated, we need to clean up the files.
+        if was_active and not version.active:
+            version.clean_resources()
+        # If the version is activated, we need to trigger a build.
+        if not was_active and version.active:
+            trigger_build(project=version.project, version=version)
+        # Purge the cache from the CDN.
+        version_changed.send(sender=self.__class__, version=version)
```
Comment (benjaoming) [inline readthedocs/api/v3/views.py:325]: That seems like an additional bug fix in the mix :+1:

File: docs/user/api/v3.rst [hunk 2]
```diff
@@ -580,6 +580,9 @@ Version update
 
     Update a version.
 
+    When a version is deactivated, its documentation is removed,
+    and when it's activated, a new build is triggered.
```
Comment (benjaoming) [inline docs/user/api/v3.rst:584]:
```suggestion
    When a version is deactivated, its documentation is removed,
    and when it's activated, a new build is triggered.
    
    Updates to versions also invalidate CDN caches.
```

File: readthedocs/api/v3/views.py [hunk 3]
```diff
@@ -297,6 +298,33 @@ def get_serializer_class(self):
...
+        was_active = version.active
+        result = super().update(request, *args, **kwargs)
+        # Get the updated version.
+        version = self.get_object()
+        # If the version is deactivated, we need to clean up the files.
+        if was_active and not version.active:
+            version.clean_resources()
+        # If the version is activated, we need to trigger a build.
+        if not was_active and version.active:
+            trigger_build(project=version.project, version=version)
```
Comment (humitos) [inline readthedocs/api/v3/views.py:323]: Is it possible to share this logic with the Form so we don't have to maintain it in both places?

Comment (benjaoming) [reply readthedocs/api/v3/views.py:323]: I'm not sure where the functionality comes from, but activating a new version also triggers a build.

Comment (stsewd) [reply readthedocs/api/v3/views.py]: @benjaoming that is done from the Version form.

@humitos I was able to unify the call from the form and the view, now this is just a call to a method from the version.

**Pull request merged**